pkg/compare/compare.go

```diff
@@ -318,7 +319,25 @@ func (o *Options) setupCorrelators() error {
 		{{"apiVersion"}, {"kind"}},
 		{{"kind"}},
 	}
-	groupCorrelator, err := NewGroupCorrelator(fieldGroups, o.templates)
+	getBestMatch := func(templates []*ReferenceTemplate, CR *unstructured.Unstructured) (*ReferenceTemplate, error) {
```

Comment:
Is it better to have this be a named function that could be separately unit-tested as opposed to an anonymous inline function?

Reply:
sounds good, moved it to a named function